Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -703,7 +710,12 @@ extension ProjectDescription.Settings {
         if !linkerFlags.isEmpty {
             settingsDictionary["OTHER_LDFLAGS"] = .array(["$(inherited)"] + linkerFlags)
         }
-
+        
+        if let settingsToOverride = targetSettings[target.name] {
+            let projectDescriptionSettingsToOverride = ProjectDescription.SettingsDictionary.from(settingsDictionary: settingsToOverride)
+            settingsDictionary.merge(projectDescriptionSettingsToOverride)
+        }
```

Comment:
What if you specify a setting which is already mapped from SPM? In case of arrays, I think we should append the values.
If `merge` already does it, I would just add one of such cases in the test you have added

Reply:
Merge does not, it overwrites. This is definitely true, but I was trying to think of principle of least astonishment here: it's really hard to tell if you should append, and it would be impossible to tell if you are appending things that conflict with each other. The safer option would be that if you are dipping into these custom dictionary values that you should make sure that you've covered the proper scenarios that would have been applied by default.

Reply:
I agree it's probably better to just override it 👍